extensions/broker-server/src/main/java/de/sovity/edc/ext/brokerserver/services/api/ConnectorApiService.java

```diff
@@ -41,8 +47,38 @@ public void addConnectors(DSLContext dsl, List<String> connectorEndpoints) {
         connectorService.addConnectors(dsl, endpoints, ADDED_ON_API_CALL);
     }
 
+    public void addConnectorsWithMdsIds(DSLContext dsl, ConnectorCreationRequest connectorCreationRequests) {
+        var connectors = connectorCreationRequests.getConnectors();
+        var existingEndpoints = connectorService.getConnectorEndpoints(dsl);
+
+        connectors.removeIf(it -> it.getConnectorEndpoint() == null || it.getMdsId() == null);
+        connectors.forEach(it -> {
+            it.setConnectorEndpoint(it.getConnectorEndpoint().trim());
+            it.setMdsId(it.getMdsId().trim());
+        });
+        connectors.removeIf(it ->
+            !UrlUtils.isValidUrl(it.getConnectorEndpoint())
+                || !MdsIdUtils.isValidMdsId(it.getMdsId())
+                || existingEndpoints.contains(it.getConnectorEndpoint())
+        );
+
+        var endpoints = connectors.stream().map(AddedConnector::getConnectorEndpoint).collect(toSet());
+        connectorService.addConnectors(dsl, endpoints, ADDED_ON_API_CALL);
+        addMdsIdsToConnectors(dsl, connectors);
```

Comment:
If a connector endpoint is contained in existingEndpoints the mdsId is not added to that connector as it is filtered out of the list beforehand. is it intended like this?

Reply:
That does not matter since a connector, which is already there, will have its MDS ID calculated from the participantId/connectorId